tools/compare_codegen.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# compare_codegen.sh - compare generated CUDA kernels between git commits
+#
+# This script is made to compare generated kernels when making changes to
+# codegen. Invoking it without any arguments while checkout origin/main, as
+# well as this commit. For each, it will build the project in release mode then
+# invoke all tests and benchmarks, saving the generated cuda kernels to files.
+# It will then diff these files and report which ones changed. The exit code is
+# the number of cuda kernels that differ between commits.
+#
+# The -r option controls the git ref to compare against. The -o option is the
+# name of the output directory.
+#
+# If the -- option is given, then instead of running all benchmarks and tests,
+# we will use the rest of the command line as the test to run. For example, to
+# compare the generated code for a single binary test, you could use:
+#
+#   tools/compare_codegen.sh -- build/nvfuser_tests --gtest_filter='*TestFoo*'
+#
+# In that case, the outputs will be placed in a subdirectory of the output
+# directory for each commit labelled "custom_command_$LAUNCHTIME" where
+# $LAUNCHTIME is a string representing the time this script was launched.
```

Comment:
It was annoying waiting four hours for results each time, so I added ability to specify a custom command. If we know a test or benchmark that might have a code change, we can see it easily this way without running everything.

Reply:
I would definitely take benchmarks out by default. If only we had a way to iterate over all scheduler configurations... 🤔

Reply:
Good idea. We can always run the benchmarks manually via
```
tools/compare_codegen.sh -- build/nvfuser_bench \
    --benchmark_repetitions=1 \
    --benchmark_min_time=0 \
    --benchmark_filter=NvFuserScheduler
```
I'll remove them from the default set of tests.

Reply:
It takes time, but the benchmarks are more important to exercise the schedulers and codegen, so we should definitely run the benchmarks. Whether it should be on by default is a different question, though.